### PR TITLE
Docs: Improved Context

### PR DESCRIPTION
**Pull request by the-r3aper7**
# What is it?

- [ ] Feature / enhancement
- [ ] Bug
- [ ] Docs / tests

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

# Use cases and why

- 1. One use case
- 2. Another use case

# Checklist:

- [ ] My code follows the [developer guidelines of this project](https://github.com/BuilderIO/qwik/blob/main/CONTRIBUTING.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

File: packages/docs/src/routes/docs/components/context/index.mdx [hunk 1]
```diff
@@ -72,34 +97,43 @@ Let's dig into every API involved:
 
 ### `createContextId()`
 
-This method takes a string that gives the context unique name, if your application uses a lot of different context or you are writing a component library, we recommend to follow a name convention that avoids conflicts, such as:
+This method is used to create a new `Context`.
```
[the remark this answers is not in the document]
Comment (the-r3aper7) [reply packages/docs/src/routes/docs/components/context/index.mdx:100]: yeah context id would be nice

File: packages/docs/src/routes/docs/components/context/index.mdx [hunk 2]
```diff
@@ -72,34 +97,43 @@ Let's dig into every API involved:
...
+
+Notice that the value returned by `createContextId()` does not hold any state, it is a immutable object i.e. `{ id: 'io.builder.qwik.city' }`. It's only used to describe the name and type of the context, like an address or an identifier.
 
-Like all use- methods, it can only be called at the root of [`component$()`](/docs/components/overview/index.mdx#component) (not inside branches). This method is called by some higher level component and it's what assigns (provides) a value to the context. The provided value will not be globally available across the whole render tree, but only to descendant components in the tree.
+Since, it does not hold any state, it's ok to call it and make it a singleton, exported in some shared module.
 
-The value passed to [`useContextProvider()`](/docs/components/context/index.mdx#usecontextprovider) can be any primitive, object (including from useStore), or array that contains serializable values.
+### `useContextProvider()`
+
+- This method is used to provide a value to `Context` after using [`createContextId()`](#createContextId).
```
Comment (shairez) [inline packages/docs/src/routes/docs/components/context/index.mdx:122]: I'd change it to :

This method is used to create a `Context` for a specific component and its descendants, using the `ContextId` as the key identifier of the context.

**Comment by shairez** [comment]
great job @the-r3aper7 !

added a few comments

File: packages/docs/src/routes/docs/components/context/index.mdx [hunk 3]
```diff
@@ -97,7 +97,7 @@ Let's dig into every API involved:
 
 ### `createContextId()`
 
-This method is used to create a new `Context`.
+This method is used to create a new `Context ID`.
```
Comment (shairez) [inline packages/docs/src/routes/docs/components/context/index.mdx:100]: Again, sorry for not making it clear  - 

I'd use etiher `ContextId` (as code)

or "Context ID" as text (wrapped with normal quotation marks)

Comment (the-r3aper7) [reply packages/docs/src/routes/docs/components/context/index.mdx:100]: i think it would be better to have "Context ID" in bold WDYT?

**Pull request merged**